packages/dds/tree/src/simple-tree/api/incrementalAllowedTypes.ts

```diff
@@ -0,0 +1,89 @@
+ * included in the summary that is uploaded to the service.
+ */
+export const incrementalAllowedTypesMetadata: AllowedTypesMetadata["custom"] = {
+	incrementalSummaryOptimization: true,
+};
+
+/**
+ * This helper function {@link getShouldIncrementallySummarizeAllowedTypes} can be used to generate a callback function
+ * of type {@link IncrementalEncodingPolicy}.
+ * This callback can be passed as the value for {@link SharedTreeOptionsInternal.shouldEncodeFieldIncrementally} parameter
+ * when creating the tree.
+ * It will be called for each allowed types in the schema to determine if it should be incrementally summarized.
+ *
+ * @param rootSchema - The schema for the root of the tree.
+ * @returns A callback function of type {@link IncrementalEncodingPolicy} which can be used to determine if a field
+ * should be incrementally summarized based on whether it is an allowed types with the
+ * {@link incrementalAllowedTypesMetadata} metadata.
+ *
+ * @remarks
+ * This only works for forest type {@link ForestTypeOptimized} and compression strategy
+ * {@link TreeCompressionStrategyExtended.CompressedIncremental}.
+ *
+ * The {@link incrementalAllowedTypesMetadata} will be replaced with a specialized metadata property once the
+ * incremental summary feature and APIs are stabilized.
+ */
+export function getShouldIncrementallySummarizeAllowedTypes(
+	rootSchema: TreeSchema,
+): IncrementalEncodingPolicy {
+	return (
+		targetNodeIdentifier: TreeNodeSchemaIdentifier | undefined,
+		targetFieldKey: FieldKey,
+	) => {
+		if (targetNodeIdentifier === undefined) {
+			// Root fields cannot be allowed types, so we don't incrementally summarize them.
+			return false;
+		}
+
+		const targetNode = rootSchema.definitions.get(targetNodeIdentifier);
+		if (targetNode === undefined) {
+			// The requested type is unknown to this schema.
+			// In this case we have no hints available from the view schema, and fall back to the default behavior of non-incremental encoding.
+			// There are two ways this can happen:
+			// 1. The view schema being used does not match the stored schema.
+			// 2. The view schema is compatible, but there are unknown optional fields which contain new types not described by the view schema.
+			return false;
+		}
+
+		if (isObjectNodeSchema(targetNode)) {
+			const targetPropertyKey = targetNode.storedKeyToPropertyKey.get(targetFieldKey);
+			if (targetPropertyKey !== undefined) {
+				const fieldSchema = targetNode.fields.get(targetPropertyKey);
+				if (fieldSchema !== undefined) {
+					return (
+						fieldSchema.allowedTypesFull.metadata.custom === incrementalAllowedTypesMetadata
+					);
+				}
+			}
+			return false;
+		}
+
+		const allowedTypes = oneFromIterable(
+			getTreeNodeSchemaPrivateData(targetNode).childAnnotatedAllowedTypes,
+		);
+		return allowedTypes === undefined
+			? false
+			: normalizeAllowedTypes(allowedTypes).metadata.custom ===
+					incrementalAllowedTypesMetadata;
+	};
```

Comment:
Non annotatedAllowedTypes can't have metadata, so you don't need to normalize this. But also, that APi should give you already normalized data.
 Fix in https://github.com/microsoft/FluidFramework/pull/25724